src/arviz_plots/plots/ppcplot.py

```diff
@@ -0,0 +1,479 @@
+            backend=backend,
+            **pc_kwargs,
+        )
+    # set reduce_dims by elimination
+    reduce_dims = [
+        dim for dim in pp_distribution.dims if dim not in set(facet_dims).union(set(sample_dims))
+    ]
+    print(f"\nreduce_dims={reduce_dims}")
+
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map.setdefault("predictive", plot_collection.aes_set)
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    # checking plot_collection wrapped dataset and viz/aes datatrees
+    print(f"\nplot_collection.data = {plot_collection.data}")
+    print(f"\nplot_collection.aes = {plot_collection.aes}")
+    print(f"\nplot_collection.viz = {plot_collection.viz}")
+
+    # ---------STEP 1 (PPC data)-------------
+    print(f"\nposterior predictive distri = {pp_distribution!r}")
+
+    # density calculation for observed variables
+    pp_kwargs = copy(plot_kwargs.get("predictive", {}))
+    print(f"\nreduce_dims = {reduce_dims!r}")
+
+    if pp_kwargs is not False:
+        _, pp_density_aes, pp_density_ignore = filter_aes(
+            plot_collection, aes_map, "predictive", reduce_dims
+        )
+        print(f"\npp_density_aes = {pp_density_aes}\npp_density_ignore= {pp_density_ignore}")
+
+        # getting first default color from color cycle and picking it
+        pp_default_color = plot_bknd.get_default_aes("color", 1, {})[0]
+        if "color" not in pp_density_aes:
+            pp_kwargs.setdefault("color", pp_default_color)
+
+        if "alpha" not in pp_density_aes:
+            pp_kwargs.setdefault("alpha", 0.2)
+
+        plot_dist_artists = ("credible_interval", "point_estimate", "point_estimate_text", "title")
+
+        print(f"\npp_kwargs = {pp_kwargs}")
+
+        # calling plot_dist with plot_collection and customized args
+        pp_dt = DataTree(name="pp_dt", data=pp_distribution)  # has to be converted from a
+        # dataarray to a datatree first before passing to plot_dist
+        plot_dist(
+            pp_dt,
+            var_names=var_names,
+            filter_vars=filter_vars,
+            group="pp_dt",
+            coords=coords,
+            sample_dims=reduce_dims,
+            kind=kind,
+            plot_collection=plot_collection,
+            labeller=labeller,
+            aes_map={
+                kind: value
+                for key, value in aes_map.items()
+                if key == "predictive" and key not in pp_density_ignore
+            },
+            plot_kwargs={
+                **{key: False for key in plot_dist_artists},
+                kind: dict(pp_kwargs.items()),
+            },
+            stats_kwargs={key: value for key, value in stats_kwargs.items() if key == "predictive"},
+        )
+
+    # ---------STEP 2 (PPC AGGREGATE)-------------
+
+    aggregate_kwargs = copy(plot_kwargs.get("aggregate", {}))
+    if aggregate and pp_kwargs is not False and aggregate_kwargs is not False:
```

Comment:
This should not depend on `pp_kwargs`, only on `aggregate` related kwargs. Also, this should be consistent with whatever happens with `observed` and `plot_kwargs["observed"]` if one raises a warning when there is duplication, the other should too, if it is an error instead, both should raise an error (ideally if an error at the top of the function before any computation or plotting happens).